src/main/java/com/github/quiram/buildhotspots/visualisation/BuildConfiguration.java

```diff
@@ -22,10 +30,19 @@
     private BuildHotspotsApplicationBase m_App = null;
     private Circle m_circle = null;
     private BuildConfigurationType m_xmlBC = null;
-
-    public BuildConfiguration(BuildConfigurationType p_xmlBC, double p_xPos, double p_yPos, BuildHotspotsApplicationBase p_App) {
+    private final ContextMenu m_contextMenu = new ContextMenu();
+    private MenuItem m_hide = new MenuItem("Hide"); //exposed as we need to dynamically show/hide this menu item when the context menu is displayed
+    
+    public BuildConfiguration(
+    		BuildConfigurationType p_xmlBC, 
+    		double p_xPos, 
+    		double p_yPos, 
+    		BuildHotspotsApplicationBase p_App
+    ) throws Exception {
         m_App = p_App;
         m_xmlBC = p_xmlBC;
+        
+        if (m_xmlBC.getBuildStats()==null) throw new Exception("There were no build stats attached to " + m_xmlBC.getName());
```

Comment:
Is it possible to make this a RuntimeException? That way we don't need to redeclare the Exception in the chain of calls.